### DISCUSSION TITLE
Which pre-built binary for Ubuntu on WSL?

### DISCUSSION THREAD

**Discussion by michaeltoohig** [nushell/nushell#10996] [Q&A] [answered]
Forgive my simple question, I've tried 3 binaries and starting to think I must need more than just extracting the downloaded .tar.gz file. I have extracted each binary, set +x on each, then tried to `./nu` or `source ./nu` but I get "cannot execute binary file". I've install the dependencies from the docs but no luck.

Just to be clear, which prebuilt binary would work on Ubuntu 22.04 on WSL? Are there known issues with WSL perhaps?

Thank you.

**Comment by fdncred** [comment]
I think this is the one you want for WSL https://github.com/nushell/nightly/releases/download/nightly-7ebae0b/nu-0.86.1-x86_64-unknown-linux-gnu.tar.gz

**Reply by michaeltoohig** [reply]
That's one that I used already without success. I've opted to install with npm and found success.

**Reply by fdncred** [reply, marked as answer]
It worked for me when I downloaded it, extracted it and did a chmod +x nu, this morning in WSL. Weird that it didn't work for you.

**Reply by michaeltoohig** [reply]
I tried on my work pc with WSL and it did work... so something just wrong on my personal machine.

**Comment by hustcer** [comment]
> That's one that I used already without success. I've opted to install with npm and found success.

npm use `x86_64-unknown-linux-musl` you can see that from https://github.com/hustcer/nu-to-npm/blob/06977f967ad0a2f2680b4f8d2f48d8bba2f7ab4a/nu/bump-ver.nu#L20

**Reply by michaeltoohig** [reply]
I'm sorry I'm not following, not sure what I'm so supposed to take from this link?
